src/shared/watchdog.c

```diff
@@ -19,52 +19,88 @@ static char *watchdog_device = NULL;
 static usec_t watchdog_timeout = USEC_INFINITY;
 static usec_t watchdog_last_ping = USEC_INFINITY;
 
-static int update_timeout(void) {
-        if (watchdog_fd < 0)
-                return 0;
-        if (watchdog_timeout == USEC_INFINITY)
-                return 0;
+static int enable_watchdog(void) {
+        int flags = WDIOS_ENABLECARD;
+
+        assert(watchdog_fd >= 0);
 
-        if (watchdog_timeout == 0) {
-                int flags;
+        if (ioctl(watchdog_fd, WDIOC_SETOPTIONS, &flags) < 0) {
+                /* ENOTTY means the watchdog is always enabled so we're fine */
+                log_full_errno(ERRNO_IS_NOT_SUPPORTED(errno) ? LOG_DEBUG : LOG_WARNING, errno,
+                                       "Failed to enable hardware watchdog, ignoring: %m");
+                if (!ERRNO_IS_NOT_SUPPORTED(errno))
+                        return -errno;
+        }
 
-                flags = WDIOS_DISABLECARD;
-                if (ioctl(watchdog_fd, WDIOC_SETOPTIONS, &flags) < 0)
-                        return log_warning_errno(errno, "Failed to disable hardware watchdog, ignoring: %m");
-        } else {
-                int sec, flags;
-                usec_t t;
+        return 0;
+}
 
-                t = DIV_ROUND_UP(watchdog_timeout, USEC_PER_SEC);
-                sec = MIN(t, (usec_t) INT_MAX); /* Saturate */
-                if (ioctl(watchdog_fd, WDIOC_SETTIMEOUT, &sec) < 0)
-                        return log_warning_errno(errno, "Failed to set timeout to %is, ignoring: %m", sec);
+static int disable_watchdog(void) {
+        int flags = WDIOS_DISABLECARD;
 
-                /* Just in case the driver is buggy */
-                assert(sec > 0);
+        assert(watchdog_fd >= 0);
 
-                /* watchdog_timeout stores the actual timeout used by the HW */
-                watchdog_timeout = sec * USEC_PER_SEC;
-                log_info("Set hardware watchdog to %s.", FORMAT_TIMESPAN(watchdog_timeout, 0));
+        if (ioctl(watchdog_fd, WDIOC_SETOPTIONS, &flags) < 0)
+                return log_warning_errno(errno, "Failed to disable hardware watchdog, ignoring: %m");
 
-                flags = WDIOS_ENABLECARD;
-                if (ioctl(watchdog_fd, WDIOC_SETOPTIONS, &flags) < 0) {
-                        /* ENOTTY means the watchdog is always enabled so we're fine */
-                        log_full_errno(ERRNO_IS_NOT_SUPPORTED(errno) ? LOG_DEBUG : LOG_WARNING, errno,
-                                       "Failed to enable hardware watchdog, ignoring: %m");
-                        if (!ERRNO_IS_NOT_SUPPORTED(errno))
-                                return -errno;
-                }
+        return 0;
+}
 
-                if (ioctl(watchdog_fd, WDIOC_KEEPALIVE, 0) < 0)
-                        return log_warning_errno(errno, "Failed to ping hardware watchdog, ignoring: %m");
+static int settimeout_watchdog(void) {
```

Comment:
I didn't follow the convention because the introduced helpers are low level things hence should never be made public like the other public functions such as `watchdog_open()`, `watchdog_close()`, etc...

Reply:
still, even then. i mean, we don't follow the rule strictly, but I think we should get this right for new code

Reply:
OK